packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/palette/DiagramPalette.tsx

```diff
@@ -39,30 +38,27 @@ const useDiagramPaletteStyle = makeStyles((theme) => ({
 }));
 
 const DiagramPaletteContent = ({ targetObjectId }: DiagramPaletteProps) => {
+  const { x: viewportX, y: viewportY, zoom: viewportZoom } = useViewport();
   const { x: paletteX, y: paletteY } = useDiagramPalette();
-  const { x, y } = usePaletteReferencePosition();
-  const { setReferencePosition, resetReferencePosition } = useLayout();
   const classes = useDiagramPaletteStyle();
 
-  const onToolApply = useCallback(() => {
-    if (x && y) {
-      setReferencePosition({ x, y }, '');
-    }
-  }, [x, y]);
+  let x: number = 0;
+  let y: number = 0;
```

Comment:
It is difficult to see the difference between `x` and `paletteX` (same for `y` and `paletteY`).
I guess consider the scale, the other one does not.